tests/freertos/main.c

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (C) 2019 Julian Holzwarth <julian.holzwarth@fu-berlin.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       freertos testing tool
+ *
+ * More detailed information about the file and the functionality implemented.
+ *
+ * @author      Julian Holzwarth <julian.holzwarth@fu-berlin.de>
+ *
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "shell.h"
+
+#include "freertos/FreeRTOS.h"
+#include"semaphore_test.h"
+
+/**
+ * Foward declarations
+ */
+static int cmd_test_mutex(int argc, char **argv);
+static int cmd_test_recursive_mutex(int argc, char **argv);
+
+
+/**
+ * @brief   List of command for this application.
+ */
+static const shell_command_t shell_commands[] = {
+    { "mutex_semaphore", "tests freertos mutex semaphor", cmd_test_mutex, },
+    { "recursive_mutex_semaphore", "tests freertos recursive mutex semaphor", cmd_test_recursive_mutex },
+    { NULL, NULL, NULL }
+};
+
+/**
+ * @brief   shell command to test freertos mutex semaphore
+ *
+ * @param[in] argc  Number of arguments
+ * @param[in] argv  Array of arguments
+ *
+ * @return 0 on success
+ */
+static int cmd_test_mutex(int argc, char **argv)
+{
+    (void) argc;
+    (void) argv;
+    puts("starting test: mutex semaphore");
+    if (semaphore_test_mutex_take() == pdPASS) {
```

Comment:
Even though this function and the command are easy to use, they don't cover all possible test cases. Maybe, there should be separate functions for creating semaphores, locking/unlocking them, and deleting them. Only with separated functions, you will be able to test functionalities from different threads.

Reply:
The test was only to test `xSemaphoreTake()` and` xSemaphoreTakeRecursive()`, with `xTicksToWait 0`. I wanted to show the bug and have a starting point for the future `freertos` library test. I hope i fixed the things you found correctly. I am sorry for the inconveniences. This is my first pr to riot.